Bulid error with latest zig version

### Zig Version

v0.12.0-dev.1808+69195d0cd 

### Zig Language Server Version

128a161b87f4538e392d0f9b9a73dcf8093e4b5d

### Steps to Reproduce

I update zig-dev-bin package on AUR and try to install zls-git package.

### Expected Behavior

It should be compiled normally.

### Actual Behavior

patching file src/config_gen/config_gen.zig
zig build-exe zls ReleaseSafe native: error: the following command failed with 4 compilation errors:
/usr/lib/zig/zig build-exe /home/Neokobase-Anstari/.cache/paru/clone/zls-git/src/zls/src/main.zig -OReleaseSafe --cache-dir /home/Neokobase-Anstari/.cache/paru/clone/zls-git/src/zls/zig-cache --global-cache-dir /home/Neokobase-Anstari/.cache/zig --name zls --mod version_data::/home/Neokobase-Anstari/.cache/paru/clone/zls-git/src/zls/zig-cache/o/40c95317b0fa0fa6ba0f991b372892e2/version_data_master_19701.zig --mod diffz::/home/Neokobase-Anstari/.cache/zig/p/122020da93490cefb22ff4b6a2a814d7572ef042e38b17f841d7195b92d69ff3be93/DiffMatchPatch.zig --mod build_options::/home/Neokobase-Anstari/.cache/paru/clone/zls-git/src/zls/zig-cache/c/cb5eecabc025c7775f85b50e26b46508/options.zig --mod known-folders::/home/Neokobase-Anstari/.cache/zig/p/122002462f37b67a54fa1ab712d870d4637dae0d94437c0077b148e1fb75616c573d/known-folders.zig --deps build_options,known-folders,diffz,version_data --listen=-
Build Summary: 3/6 steps succeeded; 1 failed (disable with --summary none)
install transitive failure
└─ install zls transitive failure
   └─ zig build-exe zls ReleaseSafe native 4 errors
src/Server.zig:697:9: error: redundant inline keyword in comptime scope
        inline for (std.meta.fields(Config), 0..) |field, index| {
        ^~~~~~
src/Server.zig:2340:5: error: redundant inline keyword in comptime scope
    inline for (types.request_metadata, &kvs_list) |meta, *kv| {
    ^~~~~~
src/Server.zig:2349:5: error: redundant inline keyword in comptime scope
    inline for (types.notification_metadata, &kvs_list) |meta, *kv| {
    ^~~~~~
src/lsp.zig:107:13: error: redundant inline keyword in comptime scope
            inline for (fields[0 .. fields.len - 1], 0..) |field, i| {
            ^~~~~~

--- a/src/Server.zig
+++ b/src/Server.zig
@@ -694,7 +694,7 @@ fn requestConfiguration(server: *Server) Error!void {
 
     const configuration_items = comptime config: {
         var comp_config: [std.meta.fields(Config).len]types.ConfigurationItem = undefined;
-        inline for (std.meta.fields(Config), 0..) |field, index| {
+        for (std.meta.fields(Config), 0..) |field, index| {
             comp_config[index] = .{
                 .section = "zls." ++ field.name,
             };
@@ -2337,7 +2337,7 @@ fn getNotificationMetadata(comptime method: []const u8) ?types.NotificationMetad
 const RequestMethodSet = blk: {
     @setEvalBranchQuota(5000);
     var kvs_list: [types.request_metadata.len]struct { []const u8 } = undefined;
-    inline for (types.request_metadata, &kvs_list) |meta, *kv| {
+    for (types.request_metadata, &kvs_list) |meta, *kv| {
         kv.* = .{meta.method};
     }
     break :blk std.ComptimeStringMap(void, &kvs_list);
@@ -2346,7 +2346,7 @@ const RequestMethodSet = blk: {
 const NotificationMethodSet = blk: {
     @setEvalBranchQuota(5000);
     var kvs_list: [types.notification_metadata.len]struct { []const u8 } = undefined;
-    inline for (types.notification_metadata, &kvs_list) |meta, *kv| {
+    for (types.notification_metadata, &kvs_list) |meta, *kv| {
         kv.* = .{meta.method};
     }
     break :blk std.ComptimeStringMap(void, &kvs_list);

--- a/src/lsp.zig
+++ b/src/lsp.zig
@@ -104,7 +104,7 @@ pub fn EnumCustomStringValues(comptime T: type, comptime contains_empty_enum: bo
             const KV = struct { []const u8, T };
             const fields = @typeInfo(T).Union.fields;
             var kvs_array: [fields.len - 1]KV = undefined;
-            inline for (fields[0 .. fields.len - 1], 0..) |field, i| {
+            for (fields[0 .. fields.len - 1], 0..) |field, i| {
                 kvs_array[i] = .{ field.name, @field(T, field.name) };
             }
             break :build_kvs kvs_array[0..];

--- a/src/stage2/AstGen.zig
+++ b/src/stage2/AstGen.zig
@@ -6307,6 +6307,9 @@ fn whileExpr(
     }
 
     const is_inline = while_full.inline_token != null;
+    if (parent_gz.is_comptime and is_inline) {
+        return astgen.failTok(while_full.inline_token.?, "redundant inline keyword in comptime scope", .{});
+    }
     const loop_tag: Zir.Inst.Tag = if (is_inline) .block_inline else .loop;
     const loop_block = try parent_gz.makeBlockInst(loop_tag, node);
     try parent_gz.instructions.append(astgen.gpa, loop_block);
@@ -6577,6 +6580,9 @@ fn forExpr(
     const need_result_rvalue = @as(LocTag, block_ri.rl) != @as(LocTag, ri.rl);
 
     const is_inline = for_full.inline_token != null;
+    if (parent_gz.is_comptime and is_inline) {
+        return astgen.failTok(for_full.inline_token.?, "redundant inline keyword in comptime scope", .{});
+    }
     const tree = astgen.tree;
     const token_tags = tree.tokens.items(.tag);
     const node_tags = tree.nodes.items(.tag);

